File: sqlitemagic-tests/app/src/main/java/com/siimkinks/sqlitemagic/transformer/MonthTransformer.java
package com.siimkinks.sqlitemagic.transformer;

import com.siimkinks.sqlitemagic.annotation.transformer.DbValueToObject;
import com.siimkinks.sqlitemagic.annotation.transformer.ObjectToDbValue;

import org.threeten.bp.Month;

public class MonthTransformer {
  @ObjectToDbValue
  public static String objectToDbValue(Month javaObject) {
    return javaObject == null ? null : javaObject.name();
  }

  @DbValueToObject
  public static Month dbValueToObject(String dbObject) {
    return dbObject == null ? null : Month.valueOf(dbObject);
  }
}

File: sqlitemagic-tests/app-kotlin/src/main/kotlin/com/siimkinks/sqlitemagic/transformer/MonthTransformer.kt
package com.siimkinks.sqlitemagic.transformer

import com.siimkinks.sqlitemagic.annotation.transformer.DbValueToObject
import com.siimkinks.sqlitemagic.annotation.transformer.ObjectToDbValue
import org.threeten.bp.Month

@ObjectToDbValue
fun objectToDbValue(javaObject: Month): String = javaObject.name

@DbValueToObject
fun dbValueToObject(dbObject: String): Month = Month
    .values()
    .firstOrNull { it.name == dbObject }
    ?: Month.JANUARY